kv/kvserver/liveness: TestScanNodeVitalityFromKV failed

kv/kvserver/liveness.TestScanNodeVitalityFromKV [failed](https://tanzanite.cluster.engflow.com/invocations/default/ba747012-b419-47c6-92a8-5759459834dc?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvbGl2ZW5lc3M6bGl2ZW5lc3NfdGVzdA==) on master @ [1e9eabb30660326665d61a2b36fe8926bc32a5f5](https://github.com/cockroachdb/cockroach/commits/1e9eabb30660326665d61a2b36fe8926bc32a5f5):


```
=== RUN   TestScanNodeVitalityFromKV
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestScanNodeVitalityFromKV1163748252
    test_log_scope.go:81: use -show-logs to present logs inline
    client_test.go:109: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness_test/pkg/kv/kvserver/liveness/client_test.go:109
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestScanNodeVitalityFromKV
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestScanNodeVitalityFromKV1163748252
--- FAIL: TestScanNodeVitalityFromKV (145.40s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #122101 kv/kvserver/liveness: TestScanNodeVitalityFromKV failed [A-testing C-bug C-test-failure O-robot P-2 T-kv branch-release-24.1 no-test-failure-activity]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScanNodeVitalityFromKV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38707

--- a/pkg/kv/kvserver/liveness/BUILD.bazel
+++ b/pkg/kv/kvserver/liveness/BUILD.bazel
@@ -49,6 +49,10 @@ go_test(
         "main_test.go",
     ],
     embed = [":liveness"],
+    exec_properties = select({
+        "//build/toolchains:is_heavy": {"test.Pool": "large"},
+        "//conditions:default": {"test.Pool": "default"},
+    }),
     deps = [
         "//pkg/base",
         "//pkg/gossip",

